src/components/media-loader.js

```diff
@@ -53,62 +55,72 @@ AFRAME.registerComponent("media-loader", {
+      const { src, index } = this.data;
 
-      this.showLoaderTimeout = this.showLoaderTimeout || setTimeout(this.showLoader, 100);
-
-      if (!url) return;
+      if (src !== oldData.src && !this.showLoaderTimeout) {
+        this.showLoaderTimeout = setTimeout(this.showLoader, 100);
+      }
 
-      const { raw, contentType } = await resolveMedia(url, token);
+      if (!src) return;
 
-      if (token) {
-        if (this.blobURL) {
-          URL.revokeObjectURL(this.blobURL);
-          this.blobURL = null;
-        }
-        const response = await fetch(raw, {
-          method: "GET",
-          headers: { Authorization: `Token ${token}` }
-        });
-        const blob = await response.blob();
-        this.blobURL = window.URL.createObjectURL(blob);
-      }
+      const { raw, images, contentType } = await resolveMedia(src, false, index);
 
-      if (contentType.startsWith("image/") || contentType.startsWith("video/") || contentType.startsWith("audio/")) {
+      const isPDF = contentType.startsWith("application/pdf");
+      if (
+        contentType.startsWith("image/") ||
+        contentType.startsWith("video/") ||
+        contentType.startsWith("audio/") ||
+        isPDF
+      ) {
+        this.el.removeAttribute("gltf-model-plus");
         this.el.addEventListener(
           "image-loaded",
-          () => {
+          async () => {
             clearTimeout(this.showLoaderTimeout);
+            delete this.showLoaderTimeout;
+            if (isPDF) {
+              const maxIndex = await fetchMaxContentIndex(src, images.png);
+              this.el.setAttribute("media-pager", { index, maxIndex });
+            }
           },
           { once: true }
         );
-        this.el.setAttribute("image-plus", { src: this.blobURL || raw, contentType, token });
-        this.el.setAttribute("position-at-box-shape-border", { target: ".delete-button", dirs: ["forward", "back"] });
+        const imageSrc = isPDF ? images.png : raw;
+        const imageContentType = isPDF ? "image/png" : contentType;
+
+        if (!isPDF) {
+          this.el.removeAttribute("media-pager");
+        }
+
+        this.el.setAttribute("image-plus", { src: imageSrc, contentType: imageContentType });
+        this.el.setAttribute("position-at-box-shape-border", { dirs: ["forward", "back"] });
       } else if (
         contentType.includes("application/octet-stream") ||
         contentType.includes("x-zip-compressed") ||
         contentType.startsWith("model/gltf") ||
-        url.endsWith(".gltf") ||
-        url.endsWith(".glb")
+        src.endsWith(".gltf") ||
+        src.endsWith(".glb")
       ) {
+        this.el.removeAttribute("image-plus");
+        this.el.removeAttribute("media-pager");
         this.el.addEventListener(
           "model-loaded",
           () => {
             clearTimeout(this.showLoaderTimeout);
             this.setShapeAndScale(this.data.resize);
+            delete this.showLoaderTimeout;
```

Comment:
swap this line and the one above it so that `delete`ing the timeout immediately follows `clear`ing the timeout.